src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/csip/CSIPNomisApiMockServer.kt

```diff
@@ -0,0 +1,232 @@
+
+    val pages = (totalElements / pageSize) + 1
+    (0..pages).forEach { page ->
+      val startId = (page * pageSize) + 1
+      val endId = min((page * pageSize) + pageSize, totalElements)
+      nomisApi.stubFor(
+        get(
+          urlPathEqualTo("/csip/ids"),
+        )
+          .withQueryParam("page", equalTo(page.toString()))
+          .willReturn(
+            aResponse().withHeader("Content-Type", "application/json").withStatus(HttpStatus.OK.value())
+              .withBody(
+                csipIdsPagedResponse(
+                  totalElements = totalElements,
+                  ids = (startId..endId).map { it },
+                  pageNumber = page,
+                  pageSize = pageSize,
+                ),
+              ),
+          ),
+      )
+    }
+  }
+
+  fun stubMultipleGetCSIP(intProgression: IntProgression) {
+    (intProgression).forEach {
+      nomisApi.stubFor(
+        get(
+          urlPathEqualTo("/csip/$it"),
+        )
+          .willReturn(
+            aResponse().withHeader("Content-Type", "application/json").withStatus(HttpStatus.OK.value())
+              .withBody(csipResponse(it.toLong())),
+          ),
+      )
+    }
+  }
+
+  fun stubGetCSIP(nomisCSIPId: Long = 1234) {
+    nomisApi.stubFor(
+      get(
+        urlPathEqualTo("/csip/$nomisCSIPId"),
+      )
+        .willReturn(
+          aResponse().withHeader("Content-Type", "application/json").withStatus(HttpStatus.OK.value())
+            .withBody(csipResponse(nomisCSIPId)),
+        ),
+    )
+  }
+
+  fun stubGetCSIPNotFound(nomisCSIPId: Long = 1234) {
+    nomisApi.stubFor(
+      get(
+        urlPathEqualTo("/csip/$nomisCSIPId"),
+      )
+        .willReturn(
+          aResponse().withHeader("Content-Type", "application/json")
+            .withStatus(HttpStatus.NOT_FOUND.value()),
+        ),
+    )
+  }
+  fun stubGetInitialCount(urlPath: String, totalElements: Long, pagedResponse: (totalElements: Long) -> String) = nomisApi.stubGetInitialCount(urlPath, totalElements, pagedResponse)
+  fun verifyGetIdsCount(url: String, fromDate: String, toDate: String, prisonId: String? = null) = nomisApi.verifyGetIdsCount(url, fromDate, toDate, prisonId)
+  fun verify(countMatchingStrategy: CountMatchingStrategy, requestPatternBuilder: RequestPatternBuilder) = nomisApi.verify(countMatchingStrategy, requestPatternBuilder)
+  fun verify(pattern: RequestPatternBuilder) = nomisApi.verify(pattern)
+  fun verify(count: Int, pattern: RequestPatternBuilder) = nomisApi.verify(count, pattern)
+}
+
+fun csipIdsPagedResponse(
+  totalElements: Long = 10,
+  ids: List<Long> = (0L..10L).toList(),
+  pageSize: Long = 10,
+  pageNumber: Long = 0,
+): String {
+  val content = ids.map { """{ "id": $it }""" }.joinToString { it }
+  return pageContent(content, pageSize, pageNumber, totalElements, ids.size)
+}
+
+private fun csipResponse(
+  nomisCSIPId: Long = 1234,
+): String =
+  """
```

Comment:
I think on balance it is better to use instances of DTO objects generated from OpenAPI then JSON in Strings

Reply:
np will update in separate PR